contract/src/main/kotlin/com/ritense/valtimo/contract/http/RestTemplateBuilderSingleton.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.valtimo.contract.http
+
+import org.springframework.boot.web.client.RestTemplateBuilder
+
+object RestTemplateBuilderSingleton {
+    private var builder: RestTemplateBuilder
+
+    fun set(builder: RestTemplateBuilder) {
```

Comment:
Hmm don't follow the set approach here

Reply:
It has to be set, we need config properties to be able to set it, so we add a setter. Could rename the method to something else, though. Any suggestions?